cancer_ai/chain_models_store.py

```diff
@@ -10,47 +10,31 @@
 class ChainMinerModel(BaseModel):
     """Uniquely identifies a trained model"""
 
-    namespace: str = Field(
-        description="Namespace where the model can be found. ex. Hugging Face username/org."
-    )
-    name: str = Field(description="Name of the model.")
-
-    epoch: int = Field(
-        description="The epoch number to submit as your checkpoint to evaluate e.g. 10"
-    )
-
-    date: datetime.datetime = Field(
-        description="The datetime at which model was pushed to hugging face"
-    )
-
-    # Identifier for competition
     competition_id: Optional[str] = Field(description="The competition id")
-
-    block: Optional[str] = Field(
-        description="Block on which this model was claimed on the chain."
-    )
-
-    hf_repo_id: Optional[str] = Field(description="Hugging Face repo id.")
-    hf_filename: Optional[str] = Field(description="Hugging Face filename.")
-    hf_repo_type: Optional[str] = Field(description="Hugging Face repo type.")
+    hf_repo_id: str | None = None
+    hf_model_filename: str | None = None
+    hf_code_filename: str | None = None
+    hf_repo_type: str | None = None
 
     class Config:
         arbitrary_types_allowed = True
 
     def to_compressed_str(self) -> str:
         """Returns a compressed string representation."""
-        return f"{self.namespace}:{self.name}:{self.epoch}:{self.competition_id}:{self.date}"
+        return f"{self.hf_repo_id}:{self.name}:{self.date}:{self.competition_id}:{self.hf_repo_type}:{self.hf_model_filename}:{self.hf_code_filename}"
```

Comment:
if we resign from date then probably self.date cannot be accessed and it will fail